model_eval.py

```diff
@@ -0,0 +1,245 @@
+        predictions = model_builder.custom_predict(
+            True, ranker, input_fn=lambda: model_builder.predict_input_fn(path))
+        rand_ranks = next(predictions)
+        count += 1
+        if count == constants._TIMEOUT:
+            print("Timeout: No random noise created")
+            model_builder.random_noise = False
+            return model_builder.ranking_array
+    model_builder.random_noise = False
+    return rand_ranks
+
+def correct_rand_noise_direction(model_builder, random_ranking_array):
+    """
+    Determines if randomly perturbed rank is perturbed in the same direction as
+    fgsm perturbation.
+
+    Args:
+      model_builder: The model in use.
+      random_ranking_array: Array of ranks based on randomly perturbed input.
+
+    Returns:
+      True if perturbation is in same direction as fgsm perturbation, False otherwise.
+    """
+    rank = model_builder.ranking_array[model_builder.answer_number]
+    rand_rank = random_ranking_array[model_builder.answer_number]
+
+    if get_fgsm_direction(model_builder) == 1:
+        if rand_rank >= rank:
+            return True
+    else:
+        if rand_rank <= rank:
+            return True
+    return False
+
+def get_fgsm_direction(model_builder):
+    """
+    Determines whether to increase or decrease ranking so that perturbed
+     question ranking approaches reference answer ranking.
+
+    Args:
+      model_builder: The model in use.
+      ranking_array: Original nonperturbed array of answer rankings.
+
+    Returns:
+      1 or -1 where 1 indicates increasing the rank of the perturbed question
+      and -1 indicates decreasing the rank of the perturbed question.
+    """
+    rank = model_builder.ranking_array[model_builder.answer_number]
+    ref_rank = model_builder.ranking_array[model_builder.reference_number]
+    if rank > ref_rank:
+        return -1
+    else:
+        return 1
+
+def reset_flags(model_builder, new_question):
+    """
+    Reset flags in the model so a new question/answer pair can
+    be evaluated/perturbed.
+
+    Args:
+      model_builder: The model in use.
+      new_question: Boolean indicating if a new question
+      should be evaluated.
+    """
+
+    if new_question:
+        model_builder.first_eval = True
+    else:
+        model_builder.first_eval = False
+
+def convert_question_num(model_builder, answer_num):
+    '''
+    Converts the question number (based on sorted answers)
+    that the user selects externally to interal question number
+
+    Args:
+      model_builder: The model in use.
+      answer_num: Answer number selected externally by user.
+
+    Returns:
+      Question number for internal use.
+    '''
+    answer_length = get_answer_size(model_builder)
+    no_padding_ranking = model_builder.ranking_array[:answer_length]
+
+    count = 0
+    rank_w_question = []
+    for rank in no_padding_ranking:
+        rank_w_question.append((count, rank))
+        count += 1

```

Comment:
`list(enumerate(a))` gives you a `(index,element)` pair representation of `a`, which is basically what your loop seems to be doing. Check for such syntactic sugar and use it wherever possible. There is a lot in python.